### ISSUE TITLE
Hash rates drop a lot today. pool.supportxmr.com

### ISSUE THREAD

**Issue by neverknoww** [cazala/coin-hive-stratum#126]
Today Im having issue with the hash rates.
Dropped a lot compare to yday.

Pool Graph: https://imgur.com/a/kaAGy
Hash Rates: https://imgur.com/a/0nLHm

Anyone having the same problem as me?

**Comment by patrickbau** [comment]
try to restart of the process

**Comment by neverknoww** [comment]
@patrickbau tried already, no luck. pm2 kill, pm2 delete .... etc
still no luck. no ideas what is wrong but you can also see their poor (supportxmr.com) also have some sort of shortage but no one confirm what is going on

how about you?

**Comment by patrickbau** [comment]
comparing to coinhive YES

**Comment by neverknoww** [comment]
@patrickbau still using coinhive? i saw you in telegram, not able to change the code yet?

**Comment by patrickbau** [comment]
Yes i'm still using coinhive in different website

**Comment by hitmanex** [comment]
you have to rest every 2 days or so or the hash rates drop (at least for me). i drop from 14kH to 1kh (windows server). might not have this issue on linux idk

**Comment by neverknoww** [comment]
@hitmanex ohhhh, interesting. I didnt know that. Usaually, right at this moment i should be hashing at 17k h/s but now it is 8.8k h/s. Are you also using the supportxmr.com?

**Comment by neverknoww** [comment]
https://imgur.com/a/63csb

look at their pool hash rates, 100% there is problem

**Comment by hitmanex** [comment]
@neverknoww yes im on supportxmr. always after a few days randomly kh drops

**Comment by patrickbau** [comment]
I also confirm, always i need to restart when i see the hashing drops

**Comment by vphelipe** [comment]
I also have the same problem, if you have many miners, use forever without logs, and run in multiple cores. Then create a crontab to restart every 6 hours.

That way it gets more stable.

**Comment by patrickbau** [comment]
@vphelipe Can you please tell how to do this ?

**Comment by vphelipe** [comment]
Yes. In path "/coin-hive-stratum/bin/" create file "cluster.js" and paste this code:

`var cluster = require('cluster');
var numCPUs = require('os').cpus().length;

if (cluster.isMaster) {
    for (var i = 0; i < numCPUs; i++) {
        cluster.fork();
    }
    cluster.on('exit', function(deadWorker, code, signal) {
        var worker = cluster.fork();
        var newPID = worker.process.pid;
        var oldPID = deadWorker.process.pid;
    });
} else {
    require("./coin-hive-stratum");
}`

And exec forever without logs:

`forever start -a -l /dev/null -o /dev/null -r /dev/null cluster.js`

Now you need install crontab on your server, and exec:

`crontab -e`

Now you paste this line and save:

`0 */6 * * * nvm use 6.9.2 && /usr/bin/forever restartall`

If you continue having difficulties with your pool, I can recommend a pool that I created for web mining, with low difficulty: https://elitexmr.com/

**Comment by vphelipe** [comment]
I do not use pm2, for some reason using it does not work correctly, use forever.

**Comment by neverknoww** [comment]
@vphelipe its all good, thanks for your info.

**Comment by hitmanex** [comment]
just use a batch file and ::start coin-hive-stratum then timeout /T 86400 then taskkill /F /IM  nodejs.exe ::start. (this is ghetto but itll work)

**Comment by letlet** [comment]
My hashrate also droped a lot about 10 hours ago, using pool.supportxmr.com too.  Shoul be some problem with pool.supportxmr.com

**Comment by vphelipe** [comment]
The problem is with that stratum and not with the pool. If you restart will work again.

The big problem with this code is that it only works with supportxmr.com and still has these constant crashes. It breaks down and does not work until restarting.

With other pools it works only for a few minutes, then drops until it reaches zero.

Still deserves many improvements, due to these chronic problems, Now I'm using [DeepMiner](https://github.com/deepwn/deepMiner) with a pool that I created for web mining ([EliteXMR](https://elitexmr.com/)) and I'm not having these problems.

**Comment by neverknoww** [comment]
@vphelipe you work with DeepMiner?

**Comment by vphelipe** [comment]
@neverknoww I am part of the team, but DeepMiner is not recommended to use with supportxmr.com, because after some time you can be banned by multiple connections. If you do not want to work with supportxmr, you can use DeepMiner in another pool of your choice.

**Comment by neverknoww** [comment]
Proof:

Pool Hash rates: https://imgur.com/a/KUfti
My hash rates: https://imgur.com/a/tt8Pi

As you can see, there are huge fall at the end of the graph and then it couldn't climb back up. The two graphs, you can see it is proportional to each other. I don't see this is our problem. Its clearly supportxmr.com have sort of problem and they are not to fix it. They capped our hash rates?

**Comment by adamdupuis** [comment]
I came across a pool that is supposed to be accepting of web miners with 100 diff: https://moneroocean.stream/ 

I tried it for a bit, but have not switched over to it yet. The other web mining pools I've come across have yet to find any blocks.

**Comment by vphelipe** [comment]
@neverknoww 

Sorry, but you're wrong. These falls in your hash rate are not normal. Here's how hash rate:

Pool Hash rates: https://imgur.com/a/6zNKZ
My hash rates: https://imgur.com/a/mLKhU

**Comment by Worldand** [comment]
@neverknoww 
which port you are using ?

**Comment by Worldand** [comment]
@neverknoww 
which port you are using

**Comment by neverknoww** [comment]
@vphelipe but you cant explain why they have spikes and falls at random time? This is also depend on the pool you are using (geo-locations and pings). Also you are not using web miner right?

@Worldand Im using 3333

**Comment by vphelipe** [comment]
@neverknoww 

My hash rate had no drop, it remained stable. With the supportxmr pool I use this stratum, just in my pool I use DeepMiner.

The big difference in my hash rate is because of the settings I've already cited.

**Comment by neverknoww** [comment]
@vphelipe Yeah, really dont know what causing the problem. If you read the first screenshots i posted, its quite obviously they are having some sort of network issues. 

Btw what is your deep miner setting? throttle? threads?

**Comment by vphelipe** [comment]
@neverknoww  With supportxmr, I use cluster.js (multiples processes), forever without logs and crontab to restart automatically every 6 hours. That's why my hash rate is stable.

**Comment by neverknoww** [comment]
@vphelipe Many thanks. I will give it a try
